samples/custom_dispatch/vulkan/shaders/udon_spec.mlir

```diff
@@ -0,0 +1,173 @@
+// Copyright 2024 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+// The configuration used for executable compilation.
+// This specifies the device configurations that support this custom kernel.
+#spirv_target = #hal.executable.target<"vulkan", "vulkan-spirv-fb", {
+  spirv.target_env = #spirv.target_env<
+    #spirv.vce<v1.3, [Shader, GroupNonUniform, GroupNonUniformArithmetic, GroupNonUniformBallot],
+                     [SPV_KHR_storage_buffer_storage_class, SPV_KHR_variable_pointers]>,
+    #spirv.resource_limits<max_compute_workgroup_size = [128, 128, 64], subgroup_size = 64>
+  >
+}>
+
+#ENCODING = array<i1: false, true>
+module attributes {transform.with_named_sequence} {
+  func.func private @argmax_1d_f32_entry_point(%arg0: tensor<1x?xf32>) -> tensor<1xi64> {
+    %c1 = arith.constant 1 : index
+    %dim = tensor.dim %arg0, %c1 : tensor<1x?xf32>
+    // Note: This is not safe if the dim size exceeds INT32_MAX. To pass a 64
+    // bit value it must be broken down into two 32-bit values for the high and
+    // low bits.
+    %dim_i32 = arith.index_cast %dim : index to i32
+    // Inline external dispatch that conforms to the ABI that the kernel
+    // requires. This is the primary reason for the surrounding function as
+    // details like tensor shape and push constants need to line up after
+    // splicing in the custom dispatch. This allows the kernel author to manage
+    // such details by hand without needing the rewrite patterns to worry about
+    // things like order of push constants.
+    %4 = hal.dispatch.extern "main"[%dim](%dim_i32, %arg0) : (i32, tensor<1x?xf32>{%dim}) -> tensor<1xi64>
+      count(%device: !hal.device, %workload: index) -> (index, index, index) {
+        %c1_0 = arith.constant 1 : index
+        hal.return %c1_0, %c1_0, %c1_0 : index, index, index
```

Comment:
So this is using `[1, 1, 1]` for the workgroup count. Is that on purpose? Can we try an example where this is not the case?

Reply:
I was planning to make another example for CPU that involved a larger subgraph to match and I can make the workgroup count non-trivial there.

Reply:
Is the SPIR-V example written for workgroup `[1, 1, 1]` ?

Reply:
yes